enterprise/coderd/prebuilds/reconcile_test.go

```diff
@@ -420,6 +426,93 @@ func TestPrebuildReconciliation(t *testing.T) {
 	}
 }
 
+func TestTemplateAdminDelete(t *testing.T) {
+	t.Parallel()
+
+	if !dbtestutil.WillUsePostgres() {
+		t.Skip("This test requires postgres")
+	}
+
+	t.Run("template admin delete prebuilds", func(t *testing.T) {
+		t.Parallel()
+
+		clock := quartz.NewMock(t)
+
+		// Setup.
+		ctx := testutil.Context(t, testutil.WaitSuperLong)
+		db, pubsub := dbtestutil.NewDB(t)
+
+		spy := newStoreSpy(db, nil)
+
+		logger := testutil.Logger(t)
+		client, _, api, owner := coderdenttest.NewWithAPI(t, &coderdenttest.Options{
+			Options: &coderdtest.Options{
+				Database: spy,
+				Pubsub:   pubsub,
+			},
+			LicenseOptions: &coderdenttest.LicenseOptions{
+				Features: license.Features{
+					codersdk.FeatureExternalProvisionerDaemons: 1,
+				},
+			},
+
+			EntitlementsUpdateInterval: time.Second,
+		})
+
+		orgID := owner.OrganizationID
+
+		provisionerCloser := coderdenttest.NewExternalProvisionerDaemon(t, client, orgID, map[string]string{
+			provisionersdk.TagScope: provisionersdk.ScopeOrganization,
+		})
+		defer provisionerCloser.Close()
+
+		reconciler := prebuilds.NewStoreReconciler(spy, pubsub, codersdk.PrebuildsConfig{}, logger, clock, prometheus.NewRegistry(), newNoopEnqueuer())
+		var claimer agplprebuilds.Claimer = prebuilds.NewEnterpriseClaimer(spy)
+		api.AGPL.PrebuildsClaimer.Store(&claimer)
+
+		version := coderdtest.CreateTemplateVersion(t, client, orgID, templateWithAgentAndPresetsWithPrebuilds(2))
+		_ = coderdtest.AwaitTemplateVersionJobCompleted(t, client, version.ID)
+		template := coderdtest.CreateTemplate(t, client, orgID, version.ID)
+		presets, err := client.TemplateVersionPresets(ctx, version.ID)
+		require.NoError(t, err)
+		require.Len(t, presets, 2)
+		preset := setupTestDBPreset(t, db, version.ID, 2, "b0rked")
+
+		templateAdminClient, _ := coderdtest.CreateAnotherUser(t, client, orgID, rbac.RoleTemplateAdmin())
+
+		workspace, _ := setupTestDBPrebuild(
+			t,
+			clock,
+			db,
+			pubsub,
+			database.WorkspaceTransitionStart,
+			database.ProvisionerJobStatusSucceeded,
+			orgID,
+			preset,
+			template.ID,
+			version.ID,
+		)
+
+		require.NoError(t, reconciler.ReconcileAll(ctx))
+
+		runningWorkspaces, err := db.GetRunningPrebuiltWorkspaces(ctx)
+		require.NoError(t, err)
+
+		prebuiltWorkspace, err := db.GetWorkspaceByID(ctx, runningWorkspaces[0].ID)
+		require.NoError(t, err)
```

Comment:
You should also require there to be at least one running workspace. This may need to be done in a `testutil.Eventually(ctx)` to avoid potential flakiness.

Reply:
Addressed in https://github.com/coder/coder/pull/18333/commits/5f896f0b3a01011695397e312cd7d59ae979dedd after moving the test to cli package. But I am not sure we need the `testutil.Eventually(ctx)` because we are inserting the data manually in the database 🤔